libpcap/inject.go

```diff
@@ -0,0 +1,168 @@
+package libpcap
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/asm"
+	"github.com/cloudflare/cbpfc"
+)
+
+/*
+Steps:
+1. Find the injection position, which is the bpf_printk call
+2. Make some necessary preparations for the injection
+3. Compile the filter expression into ebpf instructions
+4. Inject the instructions
+*/
+func InjectFilter(program *ebpf.ProgramSpec, filterExpr string) (err error) {
+	if filterExpr == "" {
+		return
+	}
+
+	injectIdx := 0
+	for idx, inst := range program.Instructions {
+		// In the kprobe_pwru.c, we deliberately put a bpf_printk call to mark the injection position, see the comments over there.
```

Comment:
GL to folks who will use bpf_printk for debugging :-) Maybe let's add a warning as a code comment in the beginning of kprobe_pwru.c?